packages/p2p-v2/src/pages/my-ads/screens/MyAds/MyAdsTable/MyAdsTable.tsx

```diff
@@ -135,6 +142,16 @@ const MyAdsTable = () => {
                     onRequestClose={onRequestClose}
                 />
             )}
+            {isShareAdsModalOpen && (
+                <ShareAdsModal
+                    id={advertId}
+                    isModalOpen={isShareAdsModalOpen}
+                    onRequestClose={() => {
+                        setIsShareAdsModalOpen(false);
+                        setAdvertId('');
+                    }}
+                />
+            )}
```

Comment:
do we need to wrap and check using isShareAdsModalOpen? shouldn't the isModalOpen prop be enough for it to determine when to show in the DOM?

Reply:
ah its because its wrapped in a fragment maybe thats why, can we add like a loader instead in the share my ads modal then you can remove the `!isLoading` check

Reply:
i did think of it.. it actually happens for all the modals.. and adding a loader will be a design change.. but there isn't any visible delay .. just added it to make sure it doesn't get rendered..

Reply:
ah yeah true true, we can ask design team about this later, to add scenarios for loading, for now we can leave this since we dont have it added yet